src/Scriban/Functions/MathFunctions.cs

```diff
@@ -25,6 +25,8 @@ namespace Scriban.Functions
 #endif
     partial class MathFunctions : ScriptObject
     {
+        private static readonly Random _random = new Random();
```

Comment:
A static object is not great here, because Scriban can be used from multiple threads. You can store the `Ramdom` in `TemplateContext.Tags`  instead.

Reply:
Ups, didn't account for multi threading. Fixed.